Closes #1075 tedge config-dir (#3)
PradeepKiruvale/thin-edge.io: docs/src/references/tedge-config-dir.md

> # `tedge --config-dir`

Reviewer: I think this title is not clear. You can discuss the title and location with Didier. I assume under "Command Line Reference" is the appropriate location, however, not really sure. 

```suggestion
# The `--config-dir` option
```

> # `tedge --config-dir`

Reviewer: Add one line break between the title and text.

> The `tedge` CLI `config-dir` option can be used to store the thin-edge

Reviewer: ```suggestion
The `tedge` CLI `--config-dir` option can be used to store the thin-edge.io
```

configuration files in a custom directory.

## Create all the directories
Create all the directories that are needed by the thin-edge for its operations in a custom path as below.

```shell

tedge --config-dir path/to/config/dir --init

> 

Reviewer: Remove the empty line here.

```
All the directories will be created in the `path/to/config/dir` directory.

Create the directories and files that are required by the `tedge_mapper`
and `tedge_agent` in a custom directory as below.

```shell

> 

Reviewer: Remove the empty line here.

tedge_mapper --config-dir path/to/config/dir --init c8y
tedge_agent --config-dir path/to/config/dir --init

```

## Manage the configuration parameters

The configuration parameters can be set/unset/list in a config file that is present
in a custom directory.

> For example, the config parameter can be set as below

Reviewer: ```suggestion
For example, the config parameter can be set as below.
```

```shell

> tedge --config-dir path/to/config/dir config set c8y.url thin-edge.io

Reviewer: This is a weird example.
```suggestion
tedge --config-dir path/to/config/dir config set c8y.url your.cumulocity.com
```

```

## Manage the certificate

This involves create/remove/upload the certificate.

> For example, the certificate can be created as below

Reviewer: ```suggestion
For example, the certificate can be created as below.
```

```shell

> 

Reviewer: Remove the empty line here.

tedge --config-dir path/to/config/dir cert create --device-id thinedge

> 

Reviewer: Remove the empty line here.

```

Now the certificate will be created in the `path/to/config/dir` directory.

> ## Tedge connect with `config-dir`

Reviewer: ```suggestion
## Connect to cloud
```

This is a two-step process

### Step 1: Update the `mosquitto.conf`

> Since the `c8y/az bridge config` files will be created in a directory given through
> `config-dir`, this path must be updated in the `mosquitto.conf` file. So, this has to be updated manually.

Reviewer: ```suggestion
Since the bridge configuration files for Cumulocity IoT or Azure IoT Hub will be created in a directory given through `--config-dir`,
the path to the bridge configuration files must be found by `mosquitto`.
So, this line has to be added to your `mosquitto.conf` file manually.
```

> `include_dir /config-dir/tedge/mosquitto-conf`

Reviewer: ```suggestion
`include_dir path/to/config/dir/tedge/mosquitto-conf`
```

### Step 2: Connect using config-dir

> Use the below command to connect to `Cumulocity or Azure` cloud using `config-dir`

Reviewer: ```suggestion
Use the below command to connect to `Cumulocity IoT or Azure IoT Hub` cloud using `--config-dir`
```
